src/Lipe/Sniffs/JS/StringConcatSniff.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Lipe\Sniffs\JS;
+
+use Lipe\Traits\EscapeOutputFunctions;
+use PHP_CodeSniffer\Util\Tokens;
+
+class StringConcatSniff extends \WordPressVIPMinimum\Sniffs\JS\StringConcatSniff
+{
+    use EscapeOutputFunctions;
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param int $stackPtr The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process_token($stackPtr)
+    {
+        $nextToken = $this->phpcsFile->findNext(Tokens::$emptyTokens, $stackPtr + 1, null, true, null, true);
```

Comment:
Actually, this is working correctly, in this case, we have to use the `sanitize` for the `HTML` element instead of the string, `StringConcat` it's looking for HTML tags, not for a string in the validation.

See: https://github.com/Automattic/VIP-Coding-Standards/blob/develop/WordPressVIPMinimum/Sniffs/JS/StringConcatSniff.php#L52

I know that the error message is pointing out the `string` instead of the `HTML` element and can be confusing, but that's the way that they handle the message 😕.

This is an example code that you can use for testing:

```
// StringConcatSniff
	const str = 'test' + sanitize( '<concat>' ) + 'test' + sanitize( '</concat>' );
```